File: challenges/anagrams/python/character_match.py
# Strips whitespace from strings
def stringStripper(string):
    stringList = []
    for char in string:
        if char != ' ':
            stringList.append(char)
    return stringList


def is_character_match(string1, string2):

    # get raw characters with no whitespace
    string1Split = stringStripper(string1.lower())
    string2Split = stringStripper(string2.lower())

    # first, check len of strings
    if len(string1Split) == len(string2Split):
        # checks characters in strings
        for char in string1Split:
            if char not in string2Split:
                return False
            else:
                return True

    # returns if strings are not equal size
    return False

File: challenges/anagrams/js/characterMatch.js
exports.isCharacterMatch = function (string1, string2) {

    // get raw characters with no whitespace
    string1list = stringStripper(string1.toLowerCase())
    string2list = stringStripper(string2.toLowerCase())

    // first, check len of strings
    if (string1list.length === string2list.length) {
        // checks characters in strings
        for (let i = 0; i < string1list.length; i++) {
            for (let j = 0; j < string2list.length; j++) {
                if (string2list.includes(string1list[i])) {
                    continue
                } else {
                    return false
                }
            }
        }
        return true
    }
    return false
}


const stringStripper = (string) => {
    result = string.split('')
    stripped = []
    for (let i = 0; i < result.length; i++) {
        if (result[i] !== ' ') {
            stripped.push(result[i])
        }
    }
    return stripped
}
